src/main/java/hudson/plugins/analysis/collector/AnalysisPublisher.java

```diff
@@ -31,113 +40,23 @@
-     * @param defaultEncoding
-     *            the default encoding to be used when reading and parsing files
-     * @param useDeltaValues
-     *            determines whether the absolute annotations delta or the
-     *            actual annotations set difference should be used to evaluate
-     *            the build stability
-     * @param unstableTotalAll
-     *            annotation threshold
-     * @param unstableTotalHigh
-     *            annotation threshold
-     * @param unstableTotalNormal
-     *            annotation threshold
-     * @param unstableTotalLow
-     *            annotation threshold
-     * @param unstableNewAll
-     *            annotation threshold
-     * @param unstableNewHigh
-     *            annotation threshold
-     * @param unstableNewNormal
-     *            annotation threshold
-     * @param unstableNewLow
-     *            annotation threshold
-     * @param failedTotalAll
-     *            annotation threshold
-     * @param failedTotalHigh
-     *            annotation threshold
-     * @param failedTotalNormal
-     *            annotation threshold
-     * @param failedTotalLow
-     *            annotation threshold
-     * @param failedNewAll
-     *            annotation threshold
-     * @param failedNewHigh
-     *            annotation threshold
-     * @param failedNewNormal
-     *            annotation threshold
-     * @param failedNewLow
-     *            annotation threshold
-     * @param isCheckStyleActivated
-     *            determines whether to collect the warnings from Checkstyle
-     * @param isDryActivated
-     *            determines whether to collect the warnings from DRY
-     * @param isFindBugsActivated
-     *            determines whether to collect the warnings from FindBugs
-     * @param isPmdActivated
-     *            determines whether to collect the warnings from PMD
-     * @param isOpenTasksActivated
-     *            determines whether to collect open tasks
-     * @param isWarningsActivated
-     *            determines whether to collect compiler warnings
-     * @param canRunOnFailed
-     *            determines whether the plug-in can run for failed builds, too
-     * @param usePreviousBuildAsReference
-     *            determines whether the previous build should be used as the
-     *            reference build
-     * @param useStableBuildAsReference
-     *            determines whether only stable builds should be used as reference builds or not
-     * @param canComputeNew
-     *            determines whether new warnings should be computed (with
-     *            respect to baseline)
+     * Default constructor.
+     * Use setter to initialize fields if needed.
      */
-    // CHECKSTYLE:OFF
-    @SuppressWarnings("PMD.ExcessiveParameterList")
     @DataBoundConstructor
-    public AnalysisPublisher(final String healthy, final String unHealthy, final String thresholdLimit,
```

Comment:
you should probably kept the old constructor but deprecated it and moved the databound annotation to the new one.